client/src/components/ItemDetails/index.jsx

```diff
@@ -0,0 +1,382 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+// import icons
+import { IconContext } from 'react-icons';
+import {
+  MdKeyboardArrowUp,
+  MdKeyboardArrowDown,
+  MdAddShoppingCart,
+  MdPlaylistAdd,
+  MdDone,
+} from 'react-icons/md';
+import {
+  FaPinterest,
+  FaFacebookF,
+  FaTwitter,
+  FaCircleNotch,
+} from 'react-icons/fa';
+
+import InputField from '../common/InputField';
+
+import './ItemDetails.css';
+
+class ItemDetails extends Component {
+  state = {
+    cart: 'default',
+    wList: 'default',
+    color: '',
+    quantity: 0,
+    tags: [],
+  };
+
+  addToCartButton = {
+    adding: (
+      <React.Fragment>
+        <FaCircleNotch className="item-details__add-to-cart--icon-spin" />{' '}
+        ADDING ..
+      </React.Fragment>
+    ),
+    added: (
+      <React.Fragment>
+        <MdDone /> ITEM ADDED
+      </React.Fragment>
+    ),
+    default: (
+      <React.Fragment>
+        <MdAddShoppingCart /> ADD TO CART
+      </React.Fragment>
+    ),
+  };
+
+  addToWListButton = {
+    adding: <FaCircleNotch className="item-details__add-to-cart--icon-spin" />,
+    added: <MdDone />,
+    default: (
+      <IconContext.Provider value={{ size: '25px' }}>
+        <MdPlaylistAdd />
+      </IconContext.Provider>
+    ),
+  };
+
+  prevItem = () => {
```

Comment:
¿Que?

I don't understand how this works.

Reply:
how what works? @FarahZaqout